File: 3512-minimum-operations-to-make-array-sum-divisible-by-k/3512-minimum-operations-to-make-array-sum-divisible-by-k.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        int sum = accumulate(nums.begin(), nums.end(), 0);
        int r = sum % k;
        int cnt = 0;
        int idx = 0;
        while (r) {
            int dif = min(nums[idx], r);
            r -= dif;
            cnt += dif;
            idx++;
        }
        return cnt;
    }
};

File: 3512-minimum-operations-to-make-array-sum-divisible-by-k/3512-minimum-operations-to-make-array-sum-divisible-by-k.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        s = sum(nums)
        r = s % k
        cnt = 0
        idx = 0
        while r:
            dif = min(r, nums[idx])
            r -= dif
            cnt += dif
            idx += 1
        return cnt 